firefox-ios/Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -468,11 +472,14 @@ class FakespotViewController: UIViewController,
         case .settingsCard:
             let view: FakespotSettingsCardView = .build()
             viewModel.settingsCardViewModel.expandState = fakespotState.isSettingsExpanded ? .expanded : .collapsed
-            viewModel.settingsCardViewModel.dismissViewController = { [weak self] action in
+            viewModel.settingsCardViewModel.dismissViewController = { [weak self] dismissPermanently, action in
                 guard let self = self, let action else { return }
-
-                store.dispatch(FakespotAction.setAppearanceTo(BoolValueContext(boolValue: false, windowUUID: windowUUID)))
-                store.dispatch(FakespotAction.surfaceDisplayedEventSend(windowUUID.context))
+                if dismissPermanently {
+                    self.triggerDismiss()
+                } else {
+                    store.dispatch(FakespotAction.setAppearanceTo(BoolValueContext(boolValue: false,
```

Comment:
I think in this case we will also have to dispatch the `surfaceDisplayedEventSend` action. Everything else looks great! Thanks @MattLichtenstein for digging into this 🙏

Reply:
@MattLichtenstein I reviewed the code for other places where we dispatch the `FakespotAction.setAppearanceTo` action with false - in no other place do we `FakespotAction.surfaceDisplayedEventSend` as well which now leads me to believe we shouldn't trigger it here also. My previous logic was that the user didn't initiate the dismissal so we shouldn't send telemetry again when restoring the view controller on iPad. But on further thought we would then have to distinguish between iPad and iPhone as well because on iPhone there is no restoration of the Fakespot modal. So for consistency with other code paths we'll keep the code like it is.
Thanks for asking all these good questions! 🙏

Reply:
@thatswinnie that makes sense, thanks for the elaborate explanation!